### PR TITLE
Meadow.OS source code integration - Part 1

### PR DESCRIPTION
**Pull request by alexischr**
## Summary

This patch starts to implement the ability to build Meadow.OS with the NuttX  mainline. It will be followed by more patches, progressively enabling full Meadow.OS functionality.

The Meadow.OS source code is needed to build this configuration. Meadow.OS is being open-sourced under the Apache 2.0 license. A prototype of the source code release and integration with NuttX mainline can be found here: https://github.com/WildernessLabs/Meadow.OS/tree/prototype

## Impact

This should only impact the Meadow F7 board configurations.

## Testing

No testing to be done, as the configuration is under development.

### CODE REVIEW

**Comment by mu578** [comment]
can you just setup a Meadow.OS project?

Meadow.OS/mk
Meadow.OS/source
Meadow.OS/nutxx/branch_version_####
Meadow.OS/diff

**Comment by alexischr** [comment]
We moved to a different model where this patch does not introduce Meadow.OS specific files or libraries like libyaml to the NuttX repo anymore. These are all moved to https://github.com/WildernessLabs/Meadow.OS/tree/prototype , which needs to be checked out side-by-side and linked with this repo for the build to function. The patch is now pretty minimal, and still enables a prototype Meadow.OS build with latest NuttX

**Comment by pkarashchenko** [comment]
I'm getting
```
mkkconfig in /Users/petro.karashchenko/eclipse-workspace/nuttx-upstream/apps/wireless
mkkconfig in /Users/petro.karashchenko/eclipse-workspace/nuttx-upstream/apps
/usr/local/bin/olddefconfig: boards/arm/stm32f7/stm32f777zit6-meadow/Kconfig:43: 'boards/arm/stm32f7/stm32f777zit6-meadow/src/Meadow.OS/espcp/Kconfig' not found (in 'source "boards/arm/stm32f7/stm32f777zit6-meadow/src/Meadow.OS/espcp/Kconfig"'). Check that environment variables are set correctly (e.g. $srctree, which is unset or blank). Also note that unset environment variables expand to the empty string.
make: *** [olddefconfig] Error 1
ERROR: failed to refresh
```
after this PR

File: boards/arm/stm32f7/stm32f777zit6-meadow/Kconfig [hunk 1]
```diff
@@ -5,4 +5,63 @@
 
 if ARCH_BOARD_MEADOW_F7MICRO
 
+config HCOM_FILESYSTEM_INIT
+    bool "Enables HCOM FS initialization support"
```
Comment (pkarashchenko) [inline boards/arm/stm32f7/stm32f777zit6-meadow/Kconfig:9]: Please use TABs in Kconfig

**Pull request merged**